### PR TITLE
yaml-cpp: Update from 0.7.0 to 0.8.0 in Mariner 3.0

### PR DESCRIPTION
**Pull request by sharath-srikanth-chellappa**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

The PR upgrades yaml-cpp package from version 0.7.0 to 0.8.0 in Mariner 3.0
###### Change Log  

- Bump version to 0.8.0 from 0.7.0
- Removing patch SPECS/yaml-cpp/yaml-cpp-static.patch since the variable been replaced in 0.8.0 with commit hash 4aad2b1 in PR#1077
- Modifying name of folder to move from %{name}-%{name}-%{version} to %{name}-%{version}
- Updating the .so files from 0.7* to 0.8*
- Changing directories of cmake/yaml-cpp and pkgconfig/yaml-cpp.pc files since they are created under /usr/lib in this version
- Changing Source to reflect the new URL format with {version}.tar.gz as opposed to yaml-cpp-{version}.tar.gz (new change in 0.8.0)

###### Does this affect the toolchain?  

**NO**
###### Test Methodology

- Pipeline run: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=494333&view=results

### CODE REVIEW

File: SPECS/yaml-cpp/yaml-cpp.signatures.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
  "Signatures": {
-  "yaml-cpp-0.7.0.tar.gz": "43e6a9fcb146ad871515f0d0873947e5d497a1c9c60c58cb102a97b47208b7c3"
+  "0.8.0.tar.gz": "fbe74bbdcee21d656715688706da3c8becfd946d92cd44705cc6098bb23b3a16"
```
Comment (PawelWMS) [inline SPECS/yaml-cpp/yaml-cpp.signatures.json:3]: Please use the previous format of the file name. Using only the version number has a high chance of causing conflicts in our sources blobstorage.

File: SPECS/yaml-cpp/yaml-cpp.spec [hunk 2]
```diff
@@ -31,7 +30,7 @@ Requires:       %{name}-devel%{?_isa} = %{version}-%{release}
 The %{name}-static package contains the static library for %{name}.
 
 %prep
-%autosetup -n %{name}-%{name}-%{version}
+%autosetup -n %{name}-%{version}
```
Comment (PawelWMS) [inline SPECS/yaml-cpp/yaml-cpp.spec:33]: If the folder name is now `%{name}-%{version}`, then that's the default and we can simplify:
```suggestion
%autosetup
```

File: SPECS/yaml-cpp/yaml-cpp.signatures.json (hunk 1 above)
Comment (PawelWMS) [reply SPECS/yaml-cpp/yaml-cpp.signatures.json:3]: And in order to keep the spec building, you'll also need to apply a change I've added for the `Source0` tag in the spec.

File: SPECS/yaml-cpp/yaml-cpp.spec [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
-Version:        0.7.0
+Version:        0.8.0
 Release:        1%{?dist}
 License:        MIT
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 URL:            https://github.com/jbeder/yaml-cpp
-Source0:        https://github.com/jbeder/yaml-cpp/archive/%{name}-%{version}.tar.gz
-Patch0:         yaml-cpp-static.patch
+Source0:        https://github.com/jbeder/yaml-cpp/archive/refs/tags/%{version}.tar.gz
```
Comment (PawelWMS) [inline SPECS/yaml-cpp/yaml-cpp.spec:9]: The mentioned update to build with a different tarball name:
```suggestion
Source0:        https://github.com/jbeder/yaml-cpp/archive/refs/tags/%{version}.tar.gz#/%{name}-%{version}.tar.gz
```

**Review by PawelWMS — CHANGES_REQUESTED** [review]
Looks good but please address the tarball name comments, since raw version names are too conflict-prone with the way our sources store is currently organized.

File: SPECS/yaml-cpp/yaml-cpp.signatures.json (hunk 1 above)
Comment (sharath-srikanth-chellappa) [reply SPECS/yaml-cpp/yaml-cpp.signatures.json]: Pipeline Run: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=494412&view=results

@PawelWMS - Does this work now and could you please take a look

**Comment by sharath-srikanth-chellappa** [comment]
> Looks good but please address the tarball name comments, since raw version names are too conflict-prone with the way our sources store is currently organized.

Pipeline run: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=494412&view=results

Have made the changes and have got a successful run

**Review by PawelWMS — APPROVED** [review]
Thanks for the updates!

**Pull request merged**